File: Leetcode/TwoSum/SolutionSortMatch.java
//time complexity: O(nlogn + n) = O(nlogn)
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int []result = {0,0};
        int []input = new int[]{0,0};
        int []inputtemp = new int[numbers.length];
        int i = 0;
        int j = numbers.length - 1;
        int temp;
        int index1=0;
        int index2=0;
        for(i = 0; i < numbers.length; i++)
        {
            inputtemp[i] = numbers[i];
        }
        java.util.Arrays.sort(numbers);
        for(i=0;i!=j;)
        {
            temp = numbers[i] + numbers[j];
            if(temp > target)
            {
                j--;  
            }else if(temp < target)
            {
                i++;  
            }else{
                input[0] = numbers[i];
                input[1] = numbers[j];
                for(i = 0; i < numbers.length; i++)
                {
                    if(inputtemp[i] == input[0])
                    {
                        index1 = i + 1;
                        break;
                    }
                }
                for(j = 0; j < numbers.length; j++)
                {
                    if(inputtemp[j] == input[1] && j != (index1-1))
                    {
                        index2 = j + 1;
                        break;
                    }
                }
                if(index1 > index2)
                {
                    result[0] = index2;
                    result[1] = index1;
                }else{
                    result[0] = index1;
                    result[1] = index2;
                }
                return result;  
            }
        }
        result[0] = 0;
        result[1] = 0;
        return result;
    }
}

File: Leetcode/TwoSum/SolutionSortMatch.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
            vector<int> out_res;
            vector<int> in_vec;
            for(int i=0;i<numbers.size();i++)
                in_vec.push_back(numbers[i]);
            std::sort(in_vec.begin(),in_vec.end());
            int i=0,j=(int)in_vec.size()-1;
            while(i<j){
                int sum = in_vec[i]+in_vec[j];
                if(sum>target){
                    j--;
                } else if(sum<target) {
                    i++;
                }else {
                    out_res.clear();
                    out_res.push_back(in_vec[i]);
                    out_res.push_back(in_vec[j]);
                    int index1=0,index2=0;
                    for(int i=0;i<numbers.size();i++)
                    {
                        if(numbers[i]==out_res[0])
                        {
                            index1=i+1;
                            break;
                        }
                    }
                    for(int i=0;i<numbers.size();i++)
                    {
                        if((numbers[i]==out_res[1])&&(i!=index1-1)){
                            index2=i+1;
                            break;
                        }
                    }
                    out_res.clear();
                    if(index1<index2){
                        out_res.push_back(index1);
                        out_res.push_back(index2);
                    }else {
                        out_res.push_back(index2);
                        out_res.push_back(index1);
                    }
                    return out_res;
                }
               
            } 
    }
};

